### PR TITLE
fix: clean up the syntax a bit

### PR DESCRIPTION
**Pull request by michaelfig**
Also, fix a bug with binary operators not being written correctly.

### CODE REVIEW

**Comment by michaelfig** [comment]
Output:

```js
** Simple Objects: stateless singleton **
const origin = {
  getX: () => 0,
  getY: () => 0
};

console.log(origin.getY());

.--------.
| maker: |
'--------'
const makeCounter = init => {
  let value = init;
  return {
    increment: () => {
      value += 1;
    },
    decrement: () => {
      value -= 1;
    },
    makeOffsetCounter: delta => makeCounter(value + delta)
  };
};

const c1 = makeCounter(1);

c1.increment();

console.log(c1.decrement());


.-------.
| mint: |
'-------'
function makeMint() {
  const ledger = makeWeakMap();
  const issuer = harden({
    makeEmptyPurse: () => {
      mint.makePurse(0);
    }
  });
  const mint = harden({
    makePurse: initialBalance => {
      const purse = harden({
        getIssuer: () => issuer,
        getBalance: () => {
          ledger.get(purse);
        },
        deposit: (amount, src) => {
          Nat(ledger.get(purse) + Nat(amount));
          ledger.set(src, Nat(ledger.get(src)) - amount);
          ledger.set(purse, ledger.get(purse) + amount);
        },
        withdraw: amount => {
          const newPurse = issuer.makeEmptyPurse();
          newPurse.deposit(amount, purse);
          return newPurse;
        }
      });
      ledger.set(purse, initialBalance);
      return purse;
    }
  });
  return mint;
}

const carolMint = makeMint();

const aliceMainPurse = carolMint.makePurse(1000);

const bobMainPurse = carolMint.makePurse(0);

const paymentForBob = aliceMainPurse.withdraw(10);

bobMainPurse.deposit(10, paymentForBob);

console.log(aliceMainPurse.getBalance());

console.log(bobMainPurse.getBalance());


.------------.
| hasher.js: |
'------------'
import {assert} from "@agoric/assert";

import {createHash} from "crypto";

function createSha256(initial = undefined) {
  const hash = createHash("sha256");
  let done = false;
  function add(more) {
    assert(not(done));
    hash.update(more);
  };
  function finish() {
    assert(not(done));
    done = true;
    return hash.digest("hex");
  };
  if (initial) {
    add(initial);
  }
  return harden({
    add: finish
  });
}

harden(createSha256);

export {create};


.---------------------------------.
| Nested math anti-infix example: |
'---------------------------------'
1 + 2 + (3 * 4) + numberFrob(42 - 2, 88) + -55;

```

File: james-to-jessie.scm [hunk 1]
```diff
@@ -258,6 +258,7 @@
       ['null (dop "null")]
       [(or 'true #t) (dop "true")]
       [(or 'false #f) (dop "false")]
+      ;; TODO: JS void is a unary operator that always evaluates to 'undefined'
```
Comment (cwebber) [inline james-to-jessie.scm:261]: Ah interesting.  In that case I wonder if the best thing to do is to wrap the following expression in parentheses, then.  That would be easy for me to add.

(I was worried about Javascript having an undefined, a null, *and* a void.  Glad that's not the case.)

File: james-to-jessie.scm [hunk 2]
```diff
@@ -276,7 +277,7 @@
        (dop " ")
        (dop (symbol->string infixer-op))
        (dop " ")
-       (write-infix-safe-expr arg1)]
+       (write-infix-safe-expr arg2)]
```
Comment (cwebber) [inline james-to-jessie.scm:280]: oops, good catch.

File: james-to-jessie.scm [hunk 3]
```diff
@@ -501,14 +504,17 @@
     ;; but it's the fastest hacky way...
     (format op "~s" expr))
 
+  ;; TODO: Named `function NAME(...args) { }` functions may end up being removed
```
Comment (cwebber) [inline james-to-jessie.scm:507]: Happy to switch to that format then.

**Pull request merged**